src/components/FlowChart/FlowSelection.jsx

```diff
@@ -1,56 +1,118 @@
-import React from "react";
+import React, { useState } from "react";
 import { Dropdown } from "react-bootstrap";
 
+function SelectionIndicator({ selectedId, itemId }) {
+  return selectedId === itemId ? (
+    <i className="material-icons text-primary">radio_button_checked</i>
+  ) : (
+    <i className="material-icons">radio_button_unchecked</i>
+  );
+}
+
+function ConnectionIndicator({ connected, show }) {
+  if (!show) return null;
+  return connected ? (
+    <i className="material-icons ml-2 text-success">wifi</i>
+  ) : (
+    <i className="material-icons ml-2 text-warning">wifi_off</i>
+  );
+}
+
```

Comment:
What was your reasoning behind doing it like that and using the component like this
```jsx        
<TitleConnectionIndicator
          show={currentFlow.type === "remote"}
          connected={connected}
        />
```

vs. not using a prop just doing
```jsx        
{currentFlow.type ==="remote" && <TitleConnectionIndicator
          connected={connected}
        />
```

At first the second version seems more intuitive to me. Why?

1. If I call a function it's me intend to do what the function name describes. If I don't want to add two numbers I better don't call `addNumbers(a, b)` and if I call it, it's my responsibility that I call it with two numbers.
I feel like the same applies for React components, which in the end are not more than just functions anyways. If I render the component it's on me to be sure, that I want to show it. Furthermore: If I need to pass the right show prop, I need anyways. Putting this on the component makes the component less flexible and adds more possible combinations of props.
2. JSX should look like the DOM created from it. If I see a component, I look for it in the DOM and it's complicated if I have to read through the props of every component to understand why it's there but not in the browser.